Net tests switch to new net_pkt/net_context APIs

tbursztyka opened the pull request:
Switching tests to the new APIs so #13618 will be merged

Commit:
tests/net: Switch checksum test to new net_context API

No need to build the net_pkt, just send the data directly through
net_context_sendto_new()

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch context test to new net_context API

No need to build the net_pkt, just send the data directly through
net_context_sendto_new()

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/pkt: Allow having NULL as iface

That can be useful on some tests which will not have any interface but
still allocate net_pkt. Also, one may allocate a packet with buffer not
knowing yet the interface it will be send through.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch icmpv6 test to new net_pkt API

No need to play with net_buf by ourselves.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch iface test to new net_pkt API

No need to play with net_buf by ourselves.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch ipv6 test to new net_pkt API

Switch to new allocators and r/w functions.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch ipv6 fragmentation test to new net_pkt API

Switch to new allocators and r/w functions.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Replace the legacy net_pkt API test by the new one.

Just a quick rm/mv. The new API is going to be the only one, so legacy
test can disappear.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

pfalcon commented:
> +813 −1,993

So, this is again, not just "switch", but also removal of more than 1000 lines of tests.

tbursztyka commented:
@pfalcon take 30 seconds next time to see the diffs and the commit messages before saying this.

pfalcon commented on tests/net/net_pkt/CMakeLists.txt:
```diff
@@ -1,7 +1,6 @@
 cmake_minimum_required(VERSION 3.13.1)
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
-project(net_pkt)
+project(NONE)
```
This isn't change for good, is it?

pfalcon commented on tests/net/net_pkt/src/main.c:
```diff
@@ -1,874 +1,617 @@
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?"
-	"0123456789abcdefghijklmnopqrstuvxyz!#¤%&/()=?";
-
-static void test_ipv6_multi_frags(void)
```
Well, this commit raises concerns that some previous testcases might be lost.

pfalcon commented:
@tbursztyka 

> @pfalcon take 30 seconds next time to see the diffs and the commit messages before saying this.

Well, I exactly took some 30 seconds for the overview of the PR, and that's what I saw. And given the case of one of the previous PRs (sorry, don't have a link handy), I had some concerns.

More detailed review took much, much more time, especially including  https://github.com/zephyrproject-rtos/zephyr/issues/14093 & https://github.com/zephyrproject-rtos/zephyr/pull/14095 . You actually could simplify review process for fellow reviewers by adding suitable summary info for the PR, e.g. "Noticeable reduction in line count is due to the fact we use net_context_send_new() which replaces a number of calls in the older API".

I still have a concern that the last commit, " tests/net: Replace the legacy net_pkt API test by the new one." might lose some useful tests. Perhaps, you could explain why it doesn't?

Commit:
tests/net: Switch MLD to new net_pkt API

Switch to new allocators and r/w functions.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Fix context test timeout values and function calls

s/0/K_NO_WAIT for all timeouts
And fix tcp context connect call (it's not net_context_listen obviously,
and one parameter was missing).

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch promiscuous test to new net_pkt API

Switch to new allocator and write function.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch tcp test to new net_pkt API

Switch to new allocators and r/w functions.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch traffic class test to new net_context API

No need to build the packet by ourselves.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: If enabled, pass the context priority to the packet

This is required if traffic class is enabled, so allocated packets from
net_context do get the right priority set.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Fix traffic class test timeout values

s/0/K_NO_WAIT for all timeouts

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/context: Add an option so set/unset packet timestamping

Though core system is able to manage packet timestamping internaly (gptp
requires it for instance), it might be necessary to enable/disable
packet timestamping from net context directly.

Currently this will be only used by the tx timestamp test. So this
support is disabled by default. (And gptp does not require it anyway).

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Switch tx timestamp test to new net_context API

No need to build the net_pkt, just send the data directly through
net_context_sendto_new()

Use the new net_context option to enable timestamping of outgoing
packet (NET_CONTEXT_TIMESTAMP).

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Reducing utils test by removing redundant ones

The very first test was basically testing the checksum calculation. And
that's already tested in more relevant tests. It was also trying in a
cumbersome way to generate packets scattered over many net_buf. But
that's also already tested in various other tests, and it's not at all
part of core utils anyway.

In any case, that all redundant, so let's remove it.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/unit: Switch vlan test to new net_pkt/net_context API

Use the new net_pkt allocator. Fix a small leak in the test as well.

No need to build the net_pkt, just send the data directly through
net_context_sendto_new()

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Fix tx timestamp test timeout values

s/0/K_NO_WAIT for timeouts

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
tests/net: Fix vlan test timeout values

s/0/K_NO_WAIT for all timeouts

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
